### DISCUSSION TITLE
GOS in MTEX

### DISCUSSION THREAD

**Discussion by wuwuhd** [ORTools4MTEX/ORTools#57] [Q&A] [answered]
Hi Frank,
I am calculating GOS  in a dual phase (mixtures of fcc asutenite and bcc martensite) steel with MTEX, but it reminds me that it is only permitted for a single phase!Could you give a coding exmaple to help me solve this problem?
best wishes!
![image](https://github.com/ORTools4MTEX/ORTools/assets/149949577/b0761629-d765-4506-916f-55822955264c)
![image](https://github.com/ORTools4MTEX/ORTools/assets/149949577/85445c32-b13f-4c08-ae95-dd0f0ee6f09a)

**Comment by AzdiarGazder** [comment, marked as answer]
Hi @wuwuhd 

From your screenshot, I believe there was an error with some of your commands.
Here is some script that works on a dual phase ebsd map.

Hope this helps.
<ins>**Lastly, and importantly, if this explanation resolves your issue, please mark this message as your answer.**</ins>

Warm regards,
Azdi


```matlab
% clear variables
clc; clear all; clear hidden; close all;

% start Mtex
startup_mtex;

% define Mtex plotting convention as X = right, Y = up
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');
setMTEXpref('FontSize', 14);

% Import the dataset
ebsd = EBSD.load('Duplex.ctf','interface','ctf',...
    'convertEuler2SpatialReferenceFrame');
ebsd = ebsd('indexed');


%% Calculate the grains
% identify grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',5*degree);
% remove small clusters
ebsd(grains(grains.grainSize <= 5)) = [];
% re-calculate grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',5*degree);
% smooth grains
grains = smooth(grains,5);

% compute GRODs
mis2mean = calcGROD(ebsd, grains);

% take the avarage of the misorientation angles for each grain
GOS =  ebsd.grainMean(mis2mean.angle);

% plot it
plot(grains, GOS ./ degree);
mtexColorbar('title','GOS in degree');
```

**Reply by wuwuhd** [reply]
Hi Azdiar，
Thanks for your help, it really works! but I still have a question  for GOS. I want to compare the GOS of two materials, and I have to set the same GOS degree range for them. can you give me some suggestion? Just like the picture below.
best wishes!
![image](https://github.com/ORTools4MTEX/ORTools/assets/149949577/eca95643-731f-4918-8f72-cd61947151d3)

**Reply by wuwuhd** [reply]
Ok, I have sloved it!

**Reply by AzdiarGazder** [reply]
Hi @wuwuhd,

Here are the steps:
1. Find the maximum GOS for the 2 maps.
   For the sake of explanation, I'm calling it "maxGOSvalue" here.

2.  Please add the following command to the end of the script in my previous reponse:
```matlab
caxis([0, maxGOSvalue]);
```

Hope this helps.
<ins>**Lastly, and importantly, if this explanation resolves your issue, please press the "Mark as answer" button on the appropriate message. Your help in this regard enables us to close discussions as complete. **</ins>

Warm regards,
Azdi

**Reply by wuwuhd** [reply]
Hi Azdiar，
It is ok! Thanks for your help!
Best wishes!

**Reply by AzdiarGazder** [reply]
Hi @wuwuhd,

I do not fully understand your question.
Is your question about the webpage you sent (on the distribution of grain boundary normals) and how to apply it to a dual phase map?
If so, then this is a new question  and is not relevant to the GOS question you raised earlier in this discussion.
So, I would be grateful if you could please delete your last reply, open a new discussion and paste it there.

While we are always happy to help new users out, please remember that we do not know your work or your data or the exact problem you are facing.
Please spend some time providing as much detail as possible when raising an issue and formulate a CLEAR question.
It really helps us understand what the issue is and provide you with a good solution quickly.

Also, instead of screenshots, I would be grateful if could please upload your script.
It saves on time re-writing it.

Looking forward to hearing from you in a new discussion! ;)

Warm regards,
Azdi

**Reply by wuwuhd** [reply]
Hi Azdiar，
it is right.! I sincerely accept your suggestion.
Best wishes!
wu
